bin/boilerplate/_extras/figures.md

```diff
@@ -5,22 +5,55 @@ title: Figures
   window.onload = function() {
     var lesson_episodes = [
     {% for episode in site.episodes %}
-    "{{ episode.url}}"{% unless forloop.last %},{% endunless %}
+    "{{ episode.url }}"{% unless forloop.last %},{% endunless %}
     {% endfor %}
     ];
+
     var xmlHttp = [];  /* Required since we are going to query every episode. */
     for (i=0; i < lesson_episodes.length; i++) {
+
       xmlHttp[i] = new XMLHttpRequest();
       xmlHttp[i].episode = lesson_episodes[i];  /* To enable use this later. */
       xmlHttp[i].onreadystatechange = function() {
+
         if (this.readyState == 4 && this.status == 200) {
-          var article_here = document.getElementById(this.episode);
           var parser = new DOMParser();
           var htmlDoc = parser.parseFromString(this.responseText,"text/html");
           var htmlDocArticle = htmlDoc.getElementsByTagName("article")[0];
-          article_here.appendChild(htmlDocArticle.getElementsByTagName("h1")[0]);
-          for (let image of htmlDocArticle.getElementsByTagName("img")) {
-            article_here.appendChild(image);
+
+          var article_here = document.getElementById(this.episode);
+          var images = htmlDocArticle.getElementsByTagName("img");
+
+          if (images.length > 0) {
+            var h1text = htmlDocArticle.getElementsByTagName("h1")[0].innerHTML;
+
+            var htitle = document.createElement('h2');
+            htitle.innerHTML = 'Episode "' + h1text + '"';
+            htitle.style = "margin-top: 100px;";
+            article_here.appendChild(htitle);
+
+            var hr = document.createElement('hr');
+            article_here.appendChild(hr);
+
+            var image_num = 0;
+            for (let image of images) {
+              image_num++;
+
+              var title = document.createElement('h3');
+              title.innerHTML = "Figure " + image_num + ". " + image.alt;
+              title.style = "margin-top: 20px;"
+              article_here.appendChild(title);
+
+              var img = document.createElement('img');
+              img.src = image.src;
+              img.style = "display: block; margin: 20px auto;"
+              article_here.appendChild(img);
+
+              if (image_num < images.length) {
+                var hr = document.createElement('hr');
```

Comment:
I don't like to have many lines between the images.

Reply:
I added these horizontal lines (and extra margins) only after looking at the final page. Without them, the title of the next image might seem related to the previous image. I'm open to suggestions how to make figure--title connection more obvious.

One possibility: images like in science papers. That is, an image, and immediately below it `Figure 01: Blah-blah-blah`